### PR TITLE
Create handler

### PR DESCRIPTION
**Pull request by orkunkl**
This branch is blocked by https://github.com/iov-one/weave/issues/924

### CODE REVIEW

**Comment by orkunkl** [comment]
Blocked by https://github.com/iov-one/weave-starter-kit/pull/31

File: x/custom/handler.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	stateIndexed := &StateIndexed{
+		Metadata: &weave.Metadata{},
+		InnerStateEnum: msg.InnerStateEnum,
+		Str: msg.Str,
+		Byte: msg.Byte,
+	}
+
+	if _, err = h.b.Create(store, stateIndexed); err != nil {
```
Comment (ruseinov) [inline x/custom/handler.go]: here you go `if obj, err = ....`

File: x/custom/handler.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		return nil, errors.Wrap(err, "cannot store indexed state")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &weave.DeliverResult{Data: stateIndexed.ID}, err
```
Comment (ruseinov) [inline x/custom/handler.go]: and here `return &weave.DeliverResult{Data: obj.Key()}, err`
And I think `stateIndexed.ID` field might be redundant. You don't need to store an ID key
within a model unless it is a reference to another model. keys and objects in our storage are stored like this `key: object`, so IDs aka keys are not really stored within the model.

Comment (orkunkl) [reply x/custom/handler.go]: The main idea was to demonstrate `morm.ModelBucket` here(which makes using models with indexes in SQL parlance easier). But then I decided to remove `morm` and use `orm.ModelBucket`. This is why I left `stateIndexed.ID` there

**Pull request merged**